runtime/runtime/src/lib.rs

```diff
@@ -1151,18 +1151,14 @@ impl Runtime {
         incoming_receipts: &[Receipt],
         transactions: &[SignedTransaction],
         epoch_info_provider: &dyn EpochInfoProvider,
-        state_patch: Option<SandboxStatePatch>,
+        state_patch: SandboxStatePatch,
     ) -> Result<ApplyResult, RuntimeError> {
         let _span = tracing::debug_span!(
             target: "runtime",
             "apply",
             num_transactions = transactions.len())
         .entered();
 
-        if state_patch.is_some() && !cfg!(feature = "sandbox") {
-            panic!("Can only patch state in sandbox mode");
-        }
```

Comment:
Let's leave `assert!(state_patch.is_empty() || cfg!(feature = "sandbox"))` as a defense in depth